client/dashboard/me/blocked-sites/index.tsx

```diff
@@ -41,13 +54,54 @@ const fields: Field< BlockedSite >[] = [
 ];
 
 export default function BlockedSites() {
-	// TODO: Implement infinite scroll once DataViews is updated to support it.
-	const { data: { sites, page } = { sites: [], page: 1 }, isLoading } = useQuery(
-		blockedSitesQuery()
-	);
+	const ref = useRef< HTMLDivElement >( null );
 	const unblockSite = useMutation( unblockSiteMutation() );
 	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
 
+	const [ view, setView ] = useState< View >( DEFAULT_VIEW );
+	const { data, isLoading, fetchNextPage, hasNextPage, isFetchingNextPage } = useInfiniteQuery(
+		blockedSitesInfiniteQuery( DEFAULT_PER_PAGE )
+	);
+
+	const sites = ( data?.pages || [] ).flat();
+	const { data: filteredData, paginationInfo } = filterSortAndPaginate( sites, view, fields );
+
+	const infiniteScrollHandler = useCallback( () => {
+		if ( hasNextPage && ! isFetchingNextPage ) {
+			fetchNextPage();
+		}
+	}, [ hasNextPage, isFetchingNextPage, fetchNextPage ] );
+
+	useEffect( () => {
+		setView( ( currentView ) => ( { ...currentView, perPage: sites.length } ) );
+	}, [ sites.length ] );
+
+	useLayoutEffect( () => {
+		if ( ! ref.current ) {
+			return;
+		}
+
+		const dataviewsRef = ref.current.querySelector< HTMLDivElement >( '.dataviews-wrapper' );
+		if ( ! dataviewsRef ) {
+			return;
+		}
+
+		const handleResize = () => {
+			const { top } = dataviewsRef.getBoundingClientRect();
+			const maxHeight = window.innerHeight - top - 32 - 1;
+			dataviewsRef.style.maxHeight = `${ maxHeight }px`;
```

Comment:
Would it be simpler to just limit the max height to the height of the 10 items 🤔

Reply:
It's simpler but it might cause double scrollbar with I'm not really a fan of 🫠